unlock-app/src/components/interface/checkout/main/Confirm/ConfirmCrossChainPurchase.tsx

```diff
@@ -100,13 +111,27 @@ export function ConfirmCrossChainPurchase({
     }
   }
 
+  useEffect(() => {
```

Comment:
I am not a huge fan of this approach because it is not very re-usable. Ideally I would want to find a way to make it easy to add support for the embedded wallet in any `ConfirmXX` screen.